src/main/src/main/java/org/geoserver/catalog/ServiceResourceVoter.java

```diff
@@ -0,0 +1,24 @@
+/* (c) 2018 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.catalog;
+
+/**
+ * Allows to verify if a given layer is suitable for publishing on a given service. A single
+ * negative vote will make the service disappear from the selective service layer UI
+ *
+ * @author Andrea Aime, Geosolutions
+ * @author Fernando Mino, Geosolutions
+ */
+public interface ServiceResourceVoter {
```

Comment:
This interface has been changed compared to the proposal, without a public discussion. This is problematic, as the proposal has been voted. Please use it in the way it has been discussed.

Reply:
Also, when going back to the original design, you'll need to add a GWC voter I believe